frontend/app/src/components/asset-manager/RestoreAssetDbButton.vue

```diff
@@ -41,21 +122,39 @@ const { restartBackend } = useBackendManagement();
 
 const { t, tc } = useI18n();
 
-const restoreAssets = async () => {
+const restoreClick = (type: ResetType) => {
+  set(resetType, type);
+  set(confirmRestore, true);
+};
+
+async function confirmReset() {
+  await restoreAssets();
+}
+
+async function restoreAssets() {
   try {
     set(confirmRestore, false);
-    set(done, await api.assets.restoreAssetsDatabase('hard', false));
+    const updated = await api.assets.restoreAssetsDatabase(
+      get(resetType),
+      get(resetType) === 'hard'
+    );
+    if (updated) {
+      set(done, true);
+    }
   } catch (e: any) {
     const title = t('asset_update.restore.title').toString();
     const message = e.toString();
+    if (message.includes('There are assets that can not')) {
```

Comment:
We probably need to think about introducing response codes when needed to communicate stuff to the frontend @rotki/rotki-devs.

Parsing the response message is not the best approach.